openedx/features/job_board/views.py

```diff
@@ -18,53 +35,76 @@ class JobListView(ListView):
     template_engine = 'mako'
 
     def get_queryset(self):
-        self.param_job_query = self.request.GET.get('query')
-        self.param_job_type = self.request.GET.getlist('type')
-        self.param_job_city = self.request.GET.get('city')
-        self.param_job_country = self.request.GET.get('country')
-        self.param_job_hours = self.request.GET.getlist('hours')
-        self.param_job_compensation = self.request.GET.getlist('compensation')
-
-        country_codes = []
-        if self.param_job_country:
-            country_codes = [country[0] for country in countries if
-                             self. param_job_country.lower() in country[1].lower()]
+        params = self.request.GET
+
+        query = params.get(JOB_PARAM_QUERY_KEY)
+        city = params.get(JOB_PARAM_CITY_KEY)
+        country = params.get(JOB_PARAM_COUNTRY_KEY)
+        remote = params.get(JOB_TYPE_CHOICES[JOB_TYPE_REMOTE_INDEX][JOB_TUPLE_KEY_INDEX])
+        onsite = params.get(JOB_TYPE_CHOICES[JOB_TYPE_ONSITE_INDEX][JOB_TUPLE_KEY_INDEX])
+        fulltime = params.get(JOB_HOURS_CHOICES[JOB_HOURS_FULLTIME_INDEX][JOB_TUPLE_KEY_INDEX])
+        parttime = params.get(JOB_HOURS_CHOICES[JOB_HOURS_PARTTIME_INDEX][JOB_TUPLE_KEY_INDEX])
+        freelance = params.get(JOB_HOURS_CHOICES[JOB_HOURS_FREELANCE_INDEX][JOB_TUPLE_KEY_INDEX])
+        volunteer = params.get(JOB_COMPENSATION_CHOICES[JOB_COMP_VOLUNTEER_INDEX][JOB_TUPLE_KEY_INDEX])
+        hourly = params.get(JOB_COMPENSATION_CHOICES[JOB_COMP_HOURLY_INDEX][JOB_TUPLE_KEY_INDEX])
+        salaried = params.get(JOB_COMPENSATION_CHOICES[JOB_COMP_SALARIED_INDEX][JOB_TUPLE_KEY_INDEX])
+
+        job_type_list = [job_type for job_type in (
+            JOB_TYPE_CHOICES[JOB_TYPE_REMOTE_INDEX][JOB_TUPLE_KEY_INDEX] if remote else [],
```

Comment:
Let's please use `None` or `''` empty list is confusing here, apart from the that it's also a falsy value. Also remote itself is not a list but it's a string.